generators/unifiedsearch-client-generator/src/test/java/uk/gov/justice/services/clients/unifiedsearch/generator/parser/UnifiedSearchDescriptorParserTest.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.justice.services.clients.unifiedsearch.generator.parser;
+
+import static java.nio.file.Paths.get;
+import static java.util.Arrays.asList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.fail;
+
+import uk.gov.justice.services.clients.unifiedsearch.generator.domain.Event;
+import uk.gov.justice.services.clients.unifiedsearch.generator.domain.UnifiedSearchDescriptor;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+
+import org.everit.json.schema.ValidationException;
+import org.junit.Test;
+
+public class UnifiedSearchDescriptorParserTest {
+
+    final Path baseDir = get("src/test/resources/yaml");
+
+    @Test
+    public void shouldFailOnIncorrectSubscriptionYaml() {
+        final UnifiedSearchDescriptorParser unifiedSearchDescriptorParser = new UnifiedSearchDescriptorFileParserFactory().create();
+
+        final List<Path> paths = asList(get("unified-search-descriptor-missing-service-component.yaml"));
+        try {
+            unifiedSearchDescriptorParser.parse(baseDir, paths);
+            fail();
+        } catch (final ValidationException re) {
+            assertThat(re, is(instanceOf(ValidationException.class)));
```

Comment:
don't need this, the exception is already caught.

Reply:
agreed and removed